source/CoopFramework/ObjectLifetimeObserver.cs

```diff
@@ -58,17 +62,43 @@ public void AfterUnregisterObject(object removedObject)
         /// <returns>Have any constructors been patched?</returns>
         public bool PatchConstruction()
         {
+            PatchLoadInitializationCallbacks();
             m_ConstructorPatch = new ConstructorPatch<ObjectLifetimeObserver<T>>(typeof(T)).PostfixAll();
             if (!m_ConstructorPatch.Methods.Any())
                 return false;
-
             foreach (var methodAccess in m_ConstructorPatch.Methods)
                 methodAccess.Postfix.SetGlobalHandler((origin, instance, args) =>
                 {
                     AfterRegisterObject(instance as T);
                 });
             return true;
         }
+        /// <summary>
+        ///     Patches the first method with <see cref="LoadInitializationCallback"/> LoadInitializationCallback attribute, but if none is found, then we check 
+        ///     if T is subclass of MBObjectBase and patch it's first method with the given attribute.. 
+        ///     This method is needed because constructors are not called at some classes, when loading a saved game, but for these classes there is a method
+        ///     in the hierarchy, that has this attribute./>.
+        /// </summary>
+        private void PatchLoadInitializationCallbacks()
+        {
+            Type type = CoopFramework.LoadInitializationCallbacks.Keys.ToList().Find(t => typeof(T) == t);
```

Comment:
Try this, you can remove one of the null checks then as well assuming it works as intended

```suggestion
            Type type = CoopFramework.LoadInitializationCallbacks.Keys.ToList().Find(t => t.IsAssignableFrom(typeof(T));
```

Reply:
Done